wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java

```diff
@@ -1077,13 +1078,18 @@ public int randomGet(Memory memory, int buf, int bufLen) {
         if (bufLen < 0) {
             return wasiResult(WasiErrno.EINVAL);
         }
-        if (bufLen >= 100_000) {
-            throw new WASMRuntimeException("random_get: bufLen must be < 100_000");
-        }
 
-        byte[] data = new byte[bufLen];
-        new SecureRandom().nextBytes(data);
-        memory.write(buf, data);
+        byte[] data = new byte[min(bufLen, 4096)];
+        int written = 0;
+        while (written < bufLen) {
+            if (Thread.currentThread().isInterrupted()) {
+                throw new ChicoryException("Thread interrupted");
+            }
+            new SecureRandom().nextBytes(data);
```

Comment:
I think that `SecureRandom` should be instantiated outside the loop.
It would be ideal to add a "Random source" to the `WasiOptions` class, I can follow up with the changes.

Reply:
Good idea! I'll add a follow up commit.

Reply:
Good idea! I'll add another commit.